pycbc/frame/frame.py

```diff
@@ -792,3 +792,112 @@ def advance(self, blocksize):
         except RuntimeError:
             self.null_advance(blocksize)
             return False
+
+class iDQBuffer(DataBuffer):
+
+    """ Read a iDQ timeseries from a frame file """
+
+    def __init__(self, frame_src,
+                 channel_name,
+                 start_time,
+                 max_buffer=512,
+                 force_update_cache=False,
+                 increment_update_cache=None):
+        """
+        Parameters
+        ----------
+        frame_src: str of list of strings
+            Strings that indicate where to read from files from. This can be a
+            list of frame files, a glob, etc.
+        channel_name: str
+            Name of the channel to read from the frame files
+        start_time:
+            Time to start reading from.
+        max_buffer: {int, 512}, Optional
+            Length of the buffer in seconds
+        force_update_cache: {boolean, True}, Optional
+            Re-check the filesystem for frame files on every attempt to
+            read more data.
+        increment_update_cache: {str, None}, Optional
+            Pattern to look for frame files in a GPS dependent directory. This
+            is an alternate to the forced updated of the frame cache, and
+            apptempts to predict the next frame file name without probing the
+            filesystem.
+        """
+        DataBuffer.__init__(self, frame_src, channel_name, start_time,
+                            max_buffer=max_buffer,
+                            force_update_cache=force_update_cache,
+                            increment_update_cache=increment_update_cache)
+
+    def lookup_idq(self, times):
+        """ Looks up the value of the idq buffer at the given times.
+
+        Parameters
+        ----------
+        times: float
+            The times whose idq values are needed
+
+        Returns
+        -------
+        values: float
+            The values of the idq buffer at the given times
+        """
```

Comment:
So it looks like the call to `at_time` is here. But this method never seems to be called? 

.... Also, I think you've said that both `times` and `values` are not floats, but numpy arrays (of floats?). Please update docstrings here (and have a quick check that others are accurate too.

Reply:
So the actual call to at_time is not in this PR. My logic was:
I am testing 3 seperate ways of implementing iDQ. They each have a seperate branch. Only of these branches will eventually be merged in. There is code that is common to all three branches. I decided to have two seperate PRs, one for the code common to all branches (this PR) and another for the statistic I decide on and code specific to supporting that statistic.

How at_time will be used is in the main pycbc_live loop, after collecting all the triggers for an ifo and removing triggers that get flagged, each trigger needs to know the idq vale at its time for that can be passed along to the single detector stat. I am doing this by pulling the array of times of the surviving triggers, passing it the iDQBuffer and asking it to give back an array with the values of idq at the time of each trigger. 

I did not know that I need to make a distinction in the docstring between a float and an array of floats. I can correct that, but I think these methods will actually work whether they are passed an array or a single value. Should the docstring say 'float or array of floats' ?